src/main/java/com/conveyal/r5/analyst/scenario/Scenario.java

```diff
@@ -113,7 +117,16 @@ public TransportNetwork applyToTransportNetwork (TransportNetwork originalNetwor
                 // could lead to meaningless errors on subsequent modifications.
                 throw new ScenarioApplicationException(Arrays.asList(modification));
             }
+
+            if (!modification.warnings.isEmpty()) {
+                modificationsWithWarnings.add(modification);
+            }
         }
+
+        copiedNetwork.scenarioApplicationWarnings = modificationsWithWarnings.stream()
+                .map(m -> new TaskError(m, m.warnings))
```

Comment:
It seems like this constructor should take one object of type Modification (since the second argument is implicit in the first), and then we should call  `modificationsWithWarnings.stream().map(TaskError::new)`